include/deal.II/grid/tria_faces.h

```diff
@@ -40,100 +40,57 @@ namespace internal
      * store the information belonging to the faces of a triangulation
      * separately from the TriaLevel classes.
      *
-     * This general template is only provided to enable a specialization
-     * below.
-     *
      * @author Tobias Leicht, 2006
+     * @author Peter Munch, 2020
      */
-
-    template <int dim>
     class TriaFaces
     {
     public:
       /**
-       * Constructor. This constructor is deleted to prevent the use of this
-       * template, as only the specializations should be used
+       * Constructor.
        */
-      TriaFaces() = delete;
-    };
-
+      TriaFaces(unsigned int dim);
 
-
-    /**
-     * Faces only have a meaning in <tt>dim@>=1</tt>. In <tt>dim=1</tt> they
-     * are vertices, which are handled differently, so only for
-     * <tt>dim@>=2</tt> the use of TriaFaces is reasonable, for <tt>dim=1</tt>
-     * the class is empty.
-     */
-    template <>
-    class TriaFaces<1>
-    {
-    public:
       /**
-       * Determine an estimate for the memory consumption (in bytes) of this
-       * object. Of course this returns 0.
+       * Default constructor for Boost::serialization.
        */
-      std::size_t
-      memory_consumption() const;
+      TriaFaces() = default;
 
       /**
-       * Read or write the data of this object to or from a stream for the
-       * purpose of serialization
+       * Dimension of the underlying triangulation.
        */
-      template <class Archive>
-      void
-      serialize(Archive &ar, const unsigned int version);
-    };
-
-    /**
-     * In <tt>dim=2</tt> the cells are quads, the faces accordingly are lines.
-     */
-    template <>
-    class TriaFaces<2>
-    {
-    public:
-      /**
-       * The TriaObject containing the data of lines.
-       */
-      TriaObjects<TriaObject<1>> lines;
+      unsigned int dim;
```

Comment:
I think you should be able to make this `const`, no?

Reply:
No, it't not possible due to serialization/deserialization of Boost.